crates/wysiwyg/src/dom/dom_struct.rs

```diff
@@ -201,11 +211,91 @@ where
         node
     }
 
-    // Return the number of code points in the string representation of this
-    // Dom.
+    /// Return the number of code points in the string representation of this
+    /// Dom.
     pub fn text_len(&self) -> usize {
         self.document.text_len()
     }
+
+    /// Add the supplied new_node into the text of the supplied handle, at
+    /// the offset supplied.
+    ///
+    /// If handle points to a text node, this text node may be split if needed.
+    /// If handle points to a line break node, offset should definitely be 1,
+    /// and the new node will be inserted after it.
+    pub fn insert_into_text(
+        &mut self,
+        handle: &DomHandle,
+        offset: usize,
+        new_node: DomNode<S>,
+    ) {
+        enum Where {
+            Before,
+            During,
+            After,
+        }
+
+        let wh = match self.lookup_node(&handle) {
+            DomNode::Container(_) => {
+                panic!("Can't insert into a non-text node!")
+            }
+            DomNode::LineBreak(_) => {
+                assert!(
+                    offset == 1,
+                    "Attempting to insert after a line break, but the offset \
+                    into it was not 1."
+                );
+                Where::After
+            }
+            DomNode::Text(n) => {
+                if offset == 0 {
+                    Where::Before
+                } else if offset == n.data().len() {
+                    Where::After
+                } else {
+                    Where::During
+                }
+            }
+        };
+
+        match wh {
+            Where::Before => {
+                self.parent(handle).insert_child(0, new_node);
```

Comment:
Shouldn't the index here be index_in_parent rather than 0 ?

Reply:
Good spot, fixed - thanks.